### ISSUE TITLE
Document how constraints affect instance selection

### ISSUE THREAD

**Issue by chexxor** [purescript/documentation#179]
It sounds like type class constraints might not affect instance selection. We should confirm that claim and add to the "language/Type-Classes.md" document.

https://purescript-users.ml/t/type-class-that-has-a-monoid-instance-seems-to-require-all-instances-to-also-be-monoids/101/8

**Comment by LiamGoodacre** [comment]
Yeah they don't contribute to comitting to an instance.  To have constraints contribute there is this ticket https://github.com/purescript/purescript/issues/3120 but it is non-trivial.
